2021/aoc/day01/solution.ml

```diff
@@ -0,0 +1,56 @@
+open List
+
+(* Reads the lines of a file into a list of strings.
+ * Taken from https://stackoverflow.com/a/23456034 *)
+let read_lines (filename : string) : string list =
+  let ic = open_in filename in
+  let try_read () = try Some (input_line ic) with End_of_file -> None in
+  let rec loop acc =
+    match try_read () with
+    | Some s -> loop (s :: acc)
+    | None ->
+        close_in ic;
+        List.rev acc
+  in
+  loop []
+
+(* Counts the number of times an element is greater than its predecessor. *)
+(* let rec count_increases (depths : int list) : int =
+  match depths with
+  | a :: b :: tl -> Bool.to_int (b > a) + count_increases (b :: tl)
+  | [ a ] -> 0
+  | [] -> 0 *)
```

Comment:
Used to use this for part1, but it was unnecessary after doing part2.  Kept for posterity as evidence of my first function in ocaml.